### PR TITLE
[docs] Paginate `/changelog`

### PR DESCRIPTION
**Pull request by keyz**
## Summary & Motivation
Can't open the https://docs.dagster.io/changelog page on my browser because it's too long.

This PR takes the `CHANGES.md` file, cuts it into chunks (while making sure each version note remains intact), and only serves a single chunk via a new `?page=` query param on the /changelog page. You can modify the `PAGINATION_VERSION_COUNT_PER_PAGE` const to specify how many versions should be shown on a page.

It assumes each version note starts with `^# \d` in `CHANGES.md`. This turns out to be good enough — no false positives. Only some very old versions (< 0.11.11) started their note with `## `, so I updated them.

Note: /changelog has been updated to a SSR page (via `getServerSideProps`), because you can't read query params in `getStaticProps`. I don't think this will be an issue (disk read is cached) but it might be worth monitoring.

cc @salazarm @yuhan

## How I Tested These Changes
image
Screenshot 2023-04-14 at 1 46 39 AM
Screenshot 2023-04-14 at 1 46 44 AM
Screenshot 2023-04-14 at 1 46 55 AM
Screenshot 2023-04-14 at 1 47 03 AM
Screenshot 2023-04-14 at 1 47 10 AM
Screenshot 2023-04-14 at 1 47 13 AM
Screenshot 2023-04-14 at 1 47 16 AM

### CODE REVIEW

**Comment by salazarm** [comment]
test

**Comment by yuhan** [comment]
vercel build here: https://dagster-8v0eu1zxg-elementl.vercel.app/changelog

File: docs/next/components/PagePagination.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    );
+  }
+
+  return (
+    <nav className="flex items-center justify-between border-t border-gray-200 px-4 sm:px-0 mt-16">
+      <div className="-mt-px flex w-0 flex-1">
+        {currentPageIndex === 0 ? null : (
+          <NextLink
+            href={{
+              query: {...query, page: currentPageIndex},
```
Comment (salazarm) [inline docs/next/components/PagePagination.tsx:102]: TIL `NextLink` has a good api

**Review by salazarm — APPROVED** [review]
Thank you! 🙏🏾

**Pull request merged**